### PR TITLE
Append TIGERA_TPROXY_ENABLED to per host envoy

### PR DESCRIPTION
**Pull request by radixo**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/applicationlayer/applicationlayer.go [hunk 1]
```diff
@@ -361,6 +361,7 @@ func (c *component) proxyEnv() []corev1.EnvVar {
 		// envoy needs to run as root to be able to use transparent flag (for tproxy)
 		{Name: "ENVOY_UID", Value: "0"},
 		{Name: "ENVOY_GID", Value: "0"},
+		{Name: "TIGERA_TPROXY_ENABLED", Value: ""},
```
Comment (tmjd) [inline pkg/render/applicationlayer/applicationlayer.go:364]: I think we should be more explicit and having a value of true or false specified or I'd like to see the name be TIGERA_TPROXY and the values of Enabled or Disabled.

Comment (radixo) [reply pkg/render/applicationlayer/applicationlayer.go:364]: This one follows as other envoy env vars are used, like "DISABLE_TEST_MERGE", where we just check for the nullptr when the envvar is not declared to go with the default behaviour of the software. In this case the TIGERA_PROXY will be always Disabled except when explicitly enabled.

Comment (tmjd) [reply pkg/render/applicationlayer/applicationlayer.go:364]: It feels ambiguous to see what is effectively `TIGERA_TPROXY_ENABLED=""`, I tried to take a look at `DISABLE_TEST_MERGE` and couldn't find it. I also tried to find envoy environment variables to see how they use environment variables to see if there was a significant amount of "prior art" with envoy environment variables. Could you share where these environment variables are defined/used?

Comment (electricjesus) [reply pkg/render/applicationlayer/applicationlayer.go:364]:
```suggestion
		{Name: "TIGERA_TPROXY", Value: "Enabled"},
```

File: pkg/render/applicationlayer/applicationlayer_test.go [hunk 2]
```diff
@@ -175,6 +175,7 @@ var _ = Describe("Tigera Secure Application Layer rendering tests", func() {
 		expectedProxyEnvs := []corev1.EnvVar{
 			{Name: "ENVOY_UID", Value: "0"},
 			{Name: "ENVOY_GID", Value: "0"},
+			{Name: "TIGERA_TPROXY_ENABLED", Value: ""},
```
Comment (electricjesus) [inline pkg/render/applicationlayer/applicationlayer_test.go:178]:
```suggestion
			{Name: "TIGERA_TPROXY", Value: "Enabled"},
```

File: pkg/render/applicationlayer/applicationlayer_test.go [hunk 3]
```diff
@@ -462,6 +463,7 @@ var _ = Describe("Tigera Secure Application Layer rendering tests", func() {
...
```
Comment (electricjesus) [inline pkg/render/applicationlayer/applicationlayer_test.go:466]:
```suggestion
			{Name: "TIGERA_TPROXY", Value: "Enabled"},
```

File: pkg/render/applicationlayer/applicationlayer_test.go [hunk 4]
```diff
@@ -620,6 +622,7 @@ var _ = Describe("Tigera Secure Application Layer rendering tests", func() {
...
```
Comment (electricjesus) [inline pkg/render/applicationlayer/applicationlayer_test.go:625]:
```suggestion
			{Name: "TIGERA_TPROXY", Value: "Enabled"},
```

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**